### PR TITLE
generate templates to support passing in package, language

### PR DESCRIPTION
**Pull request by skizhak**
- package and language uses 'gen' and 'go' by default. you can override them in the template yaml passed in.
- removed using template option under generate command. use gohan template option directly

### CODE REVIEW

File: cli/generate.go [hunk 1]
```diff
@@ -28,9 +30,9 @@ func getGenerateCommand() cli.Command {
 		ShortName: "gen",
 		Usage:     "Generate ServerSide Code",
 		Flags: []cli.Flag{
-			cli.StringFlag{Name: "template, t", Value: "", Usage: "Application template path"},
 			cli.StringFlag{Name: "templates", Value: "", Usage: "Template Configuraion"},
```
Comment (nati) [inline cli/generate.go]: could you remove template option?

File: cli/generate.go [hunk 2]
```diff
@@ -178,17 +185,11 @@ func gohanGenerate(c *cli.Context) {
...
-				"goimports -w %s/*.go", codeDir))
-	}
-	if template != "" {
-		execCommand(
-			fmt.Sprintf(
-				"gohan template --config-file %s --template %s | grep -v '^\\s*$' > %s/base_controller.go", configFile, template, packageName))
-		execCommand(
-			fmt.Sprintf(
-				"goimports -w %s/base_controller.go", codeDir))
+		if language == "go" {
```
Comment (nati) [inline cli/generate.go]: this should be in the loop.

Comment (skizhak) [reply cli/generate.go]: yes, updated.

File: cli/generate.go (hunk 1 above)
Comment (skizhak) [reply cli/generate.go]: template is already removed..

Comment (nati) [reply cli/generate.go]: oh ok thx

**Pull request merged**